src/ser-ext-ondemandDirective.ts

```diff
@@ -660,74 +668,77 @@ class OnDemandController implements ng.IController {
-                        this.state = ESERState.serNotRunning;
-                        return;
-                    }
-                } catch (error) {
-                    this.logger.error("ERROR", error);
-                    return;
-                }
+        try {
+            var response = await this.model.app.evaluate(serCall);
+            let statusObject: ISERResponseStatus;
+            this.logger.debug("response from status call", response);
 
-                try {
-                    statusObject = JSON.parse(response);
-                } catch (error) {
-                    this.logger.error("Error log from SER: ", response);
-                    this.state = ESERState.error;
+            try {
+                if (response.indexOf("Error in expression") !== -1) {
+                    this.logger.warn(response);
+                    this.state = ESERState.serNotRunning;
+                    return true;
                 }
+            } catch (error) {
+                this.logger.error("ERROR", error);
+                return true;
+            }
 
-                if (typeof (statusObject.taskId) !== "undefined") {
-                    this.taskId = statusObject.taskId;
-                }
+            try {
+                statusObject = JSON.parse(response);
+            } catch (error) {
+                this.logger.error("Error log from SER: ", response);
+                this.state = ESERState.error;
+                return true
+            }
 
-                this.logger.debug("statusObject.Status", statusObject.status);
+            if (typeof (statusObject.taskId) !== "undefined") {
+                this.taskId = statusObject.taskId;
+            }
 
-                switch (statusObject.status) {
-                    case -2:
-                        this.state = ESERState.serNoConnectionQlik;
-                        break;
-                    case -1:
-                        this.state = ESERState.error;
-                        break;
-                    case 0:
-                        this.state = ESERState.ready;
-                        this.logger.info("SER Status is ready");
-                        break;
-                    case 1:
-                        this.state = ESERState.running;
-                        break;
-                    case 2:
-                        this.state = ESERState.running;
-                        break;
-                    case 3:
-
-                        let distributeObject: IDistribute = JSON.parse(statusObject.distribute);
+            this.logger.debug("statusObject.Status", statusObject.status);
 
-                        this.links = [];
-                        for (const hubResult of distributeObject.hubResults) {
-                            if (hubResult.success) {
-                                this.links.push(`${this.host}${hubResult.link}`)
-                            }
+            switch (statusObject.status) {
+                case -2:
```

Comment:
please implement enum for status, so it is better to read. Now u know -2 means error but if u come back a half year later the first question will be "What is -2 ?" 

```
export const enum RESPONSE_STATUS {
    ERROR: -2,
    ...
    COMPLETED: 3
}

// and then implement this in switch case
switch (statusObject.status) {
    case ResponseStatus.ERROR: ... break;
```

Reply:
is included in new commit